django_app/redbox_app/redbox_core/models.py

```diff
@@ -51,22 +51,54 @@ def sanitise_string(string: str | None) -> str | None:
     return string.replace("\x00", "\ufffd") if string else string
 
 
+class ChatLLMBackend(models.Model):
```

Comment:
So this is where your approach slightly differed from my own. I had split the model creation from the available models via the `AiSettings`. The idea of this would be that you could have different AiSettings for different users could mean they have different models registered, stored in the through table `RegisteredModels`. Maybe it is over engineering it at this point, but happy to hear what you think?

Reply:
An idea I am also less cold on as this integrate with chat_init_model, is maintaining Embedding models in the same table. This could mean for an application you can possibly in the future have different config with different embedding models and registered indicies. Possibly this is a nice direction for a tests framework, to run comaprisions and record results. Maybe it doesn't live in redbox either and is a seperate thing.

Reply:
> different AiSettings for different users could mean they have different models registered

I would prefer to keep this as simple as possible, would my proposal be acceptable for now and we can revisit this in the near future? 

> maintaining Embedding models in the same table

I like this idea, it is the natural next step. Please could we add this as a separate PR?

Reply:
That's fair and happy to go with the simpler option first and change as needed. 

Makes sense to me, I thought trying to do the embedding model here may have been too much.